apps/devhub/views.py

```diff
@@ -1736,6 +1736,7 @@ def api_key_agreement(request):
     return render_agreement(request, 'devhub/api/agreement.html', next_step)
 
 
+@login_required
```

Comment:
move this decorator to `api_key_agreement` instead because `render_agreement` is not used directly as a request handler. You always want to protect the highest level view in case a view were to do something like this (just as an example) : 
````
def some_view(request):
    form = SomeForm(request)
    if form.is_valid():
        return render_agreement(...)
````